sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mlme.c

```diff
@@ -307,24 +309,45 @@ static void _mlme_link_check_ans(gnrc_lorawan_t *mac, uint8_t *p)
     gnrc_lorawan_mlme_confirm(mac, &mlme_confirm);
 
     mac->mlme.pending_mlme_opts &= ~GNRC_LORAWAN_MLME_OPTS_LINK_CHECK_REQ;
+
+    return GNRC_LORAWAN_FOPT_LINK_CHECK_ANS_SIZE;
 }
 
-static void _mlme_link_adr_req(gnrc_lorawan_t *mac, uint8_t *p)
+static int _mlme_link_adr_req(gnrc_lorawan_t *mac, uint8_t *p, size_t len, uint8_t index)
 {
-    DEBUG("gnrc_lorawan_mlme: LinkADRReq DataRate_TXPower : DR%u TX%u\n",
-           p[1] >> 4, p[1] & 0x0f);
-    DEBUG("gnrc_lorawan_mlme: LinkADRReq ChMask : %u\n", (p[3] << 8 ) | p[2]);
-    DEBUG("gnrc_lorawan_mlme: LinkADRReq Redundancy : %u\n", p[4]);
+    uint8_t last_adr_pos;
+
+    for (; index < len; index += GNRC_LORAWAN_FOPT_LINK_ADR_REQ_SIZE ) {
+        if ( p[index] == GNRC_LORAWAN_CID_LINK_ADR_REQ )
+        {
+            mac->mlme.adr_req_cnt++;
+        }
+    }
```

Comment:
this block fails to detect if there's a non-contiguous atomic block. It should be adapted to only detect the first atomic block.

Reply:
As discussed offline , In case of  `LinkAdrReq1 - LinkAdrReq2 - AnotherCommand - LinkAdrReq3`. The idea was  to accommodate `LinkAdrReq3`. Since the standard is vague and Semtech stack rejects it, we may discard the third one. Shall correct it.

Reply:
Done